service-oriented-architecture/src/main/java/com/iluwatar/soa/services/classes/PersonalizedGreetingServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+package com.iluwatar.soa.services.classes;
+
+import com.iluwatar.soa.model.WeatherCondition;
+import com.iluwatar.soa.services.interfaces.GreetingService;
+import com.iluwatar.soa.services.interfaces.PersonalizedGreetingService;
+import com.iluwatar.soa.services.interfaces.WeatherService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class PersonalizedGreetingServiceImpl implements PersonalizedGreetingService {
+
+  private final GreetingService greetingService;
+  private final WeatherService weatherService;
+
+  @Autowired
+  public PersonalizedGreetingServiceImpl(GreetingService greetingService, WeatherService weatherService) {
+    this.greetingService = greetingService;
+    this.weatherService = weatherService;
+  }
+
+  public String generateGreeting() {
+    String weatherGreeting = getWeatherGreeting();
+    return weatherGreeting + "! " + greetingService.getGenericGreeting();
+  }
+
+  private String getWeatherGreeting() {
+    WeatherCondition currentWeather = weatherService.getCurrentWeather();
+    switch (currentWeather) {
```

Comment:
Switch expressions can make this cleaner 

Something like that: 

return switch (currentWeather) {
            case SUNNY -> "What a good sunny day";  
            case RAINY -> "What a rainy day"; 
            ...
            ...
            default -> "Hello, and hope you have a great day despite the weather"; 
        };